triggers/createFile.js

```diff
@@ -0,0 +1,252 @@
+var hookID = "";
+const eventIds = [];
+
+const subscribeHook = async (z, bundle) => {
+	const { v4: uuidv4 } = require("uuid");
+	const zapier = require("zapier-platform-core");
+	zapier.tools.env.inject();
+	const eventIds = [];
+
+	const fetchEvents = {
+		url: `${process.env.BASE_URL}/service/platform/notification/v1.0/events`,
+		method: "GET",
+	};
+
+	try {
+		let response = await z.request(fetchEvents);
+
+		if (response.status === 200) {
+			const temp = [...response.data];
+
+			const obj = temp.find(
+				(item) => item.name === "file" && item.type === "create"
+			);
+			eventIds.push(obj._id);
+		} else {
+			throw new Error(`Failed to retrieve events. Status: ${response.status}`);
+		}
+	} catch (error) {
+		z.console.log("Error fetching events: " + error.message);
+		throw error;
+	}
+
+	const testWebHook = {
+		url: `${process.env.BASE_URL}/service/platform/notification/v1.0/webhook-configs/test`,
+		method: "POST",
+		body: {
+			url: "https://www.example.com",
+			secret: "",
+		},
+	};
+
+	try {
+		let testHookResponse = await z.request(testWebHook);
+		if (testHookResponse.status === 200) {
+			try {
+				const webhookConfigResponse = await z.request({
+					url: `${process.env.BASE_URL}/service/platform/notification/v1.0/webhook-configs`,
+					method: "POST",
+					body: {
+						events: [...eventIds],
+						isActive: true,
+						name: uuidv4(),
+						secret: "",
+						url: bundle.targetUrl,
+					},
+				});
+
+				if (webhookConfigResponse.status === 200) {
+					hookID = webhookConfigResponse.data.webhookConfigId;
+					return webhookConfigResponse.data;
+				} else {
+					throw new Error(
+						`Failed to create webhook configuration. Status: ${webhookConfigResponse.status}`
+					);
+				}
+			} catch (error) {
+				z.console.log("Error creating webhook configuration: " + error.message);
+				throw error;
+			}
+		}
+	} catch (error) {
+		z.console.log("Error creating test WEBHOOK: " + error.message);
+		throw new Error(
+			`Failed to create a test webhook configuration. Status: ${error}`
+		);
+	}
+};
+
+const performList = async (z, bundle) => {
```

Comment:
write a comment to explain what is happening here

Reply:
resolved